pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -192,7 +192,17 @@ def restore_training_state(self) -> None:
             return
 
         # restore precision plugin (scaler etc.)
-        self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
+        prec_plugin = self.trainer.precision_plugin
+        prec_plugin.on_load_checkpoint(self._loaded_checkpoint)
+        if prec_plugin.__class__.__name__ in self._loaded_checkpoint:
+            prec_plugin.load_state_dict(self._loaded_checkpoint[prec_plugin.__class__.__name__])
+
+        # old checkpoints compatibility
+        # should we raise error and force user to run utilities/upgrade_checkpoint instead?
```

Comment:
see comment

Reply:
cc @awaelchli - we must keep backwards compatibility with old checkpoints. if we can do this best effort with comments about including new keys & falling back to old keys for precision-related logic, we should do that.

Reply:
Making users upgrade checkpoints is tedious. Let's avoid it for as long as possible.

Reply:
>  should we raise error and force user to run utilities/upgrade_checkpoint instead?

Neither IMO. Lightning should upgrade the checkpoint directly when loading the file, and before sending the dict to the internal components. The development of this feature is currently stale: #9396
Here for example, an upgrade would consist of detecting the keys amp_scaling_state, native_amp_scaling_state, and merging them into the new format saved under the new key `NativeMixedPrecisionPlugin.__class.__name__` or whatever it ends up being :)

Reply:
I agree. For now, we can keep it as it is and refactor with #9396.

```suggestion
        # old checkpoints compatibility
```